### ISSUE TITLE
Support `/c` as alias for `-c` argument on Windows

### ISSUE THREAD

**Issue by Victor-N-Suadicani** [nushell/nushell#7993]
### Related problem

On Windows, if I set the `ComSpec` environment variable (which specifies the command interpreter) to `C:\Users\Victor\.cargo\bin\nu.exe`, and then call [libc::system](https://docs.rs/libc/latest/libc/fn.system.html) ([Microsoft docs](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/system-wsystem?view=msvc-170)) with whatever given command, I get the following error message:
```
Error: nu::shell::file_not_found (link)

  × File not found
   ╭─[source:1:1]
 1 │ nu
   · ▲
   · ╰── Could not access file '/c': "The system cannot find the file specified. (os error 2)"
   ╰────
```
I suspect this happens because the `libc::system` call expects the command interpreter to respond to the `/c` flag by running the given command. Unfortunately, nushell interprets the `/c` as a script file argument and attempts to open it, resulting in the above error message.

### Describe the solution you'd like

Support `/c` as an alias for the `-c` argument on Windows (or all platforms, if that's easier).

**Comment by Victor-N-Suadicani** [comment]
That's understandable - could you point me in the right direction for where this argument parsing occurs? I could try my hand at a PR.

**Comment by rgwood** [comment]
Sure thing. You'll want to take a look at [`src/command.rs`](https://github.com/nushell/nushell/blob/main/src/command.rs).

`gather_commandline_args()` and `parse_commandline_args()` from `command.rs` get called here in `main.rs`:
https://github.com/nushell/nushell/blob/a562f492e385fa2007cec0064f8ac3d5550372ce/src/main.rs#L62-L63

**Comment by Victor-N-Suadicani** [comment]
As for the context, basically I was writing a CLI tool that would wrap another command, e.g. `run_as_a_child "cargo help"` and I noticed that the command would always be executed in `cmd.exe`. Indeed, even when `ComSpec` is set to something else, it seems `std::process::Command` launches `cmd.exe` regardless. So then I tried the raw system call with `libc` and that actually used `ComSpec` but then I got the error when I tried with nushell.

**Comment by ChrisDenton** [comment]
Hmm... `std::process::Command` shouldn't launch cmd.exe unless you're running a `.bat` or `.cmd` file (and even in that case it's better to run the shell explicitly because the rust std's support for this is rudimentary).

`libc::system` is higher level than rust's std and does check `ComSpec` itself. Can your CLI tool not do the same?

**Comment by Victor-N-Suadicani** [comment]
@ChrisDenton I tried it out just running something like this `Command::new("help").status().unwrap()` and that shows the cmd.exe help. But you're right that it doesn't run cmd.exe for all commands it seems.

>`libc::system` is higher level than rust's std and does check `ComSpec` itself. Can your CLI tool not do the same?

What would you suggest I do after I check `ComSpec`? Run the command with `/c` or `-c`? I would want it to work with any shell the user has specified in `ComSpec`, not just nushell. I would prefer not to write specific logic just for nushell (or any other shell).

**Comment by ChrisDenton** [comment]
I don't think that `help` is using `cmd.exe`, `Command` is running `C:\Windows\System32\help.exe` which prints to an empty console without a shell at all.

---

Note that `libc::system` isn't doing anything special either. It's assuming `ComSpec` contains `cmd.exe` and acting accordingly. Maybe other shells work with that or maybe not. It depends on both whether they understand the `/c` argument and if they parse the command line the same way (e.g. cmd.exe handles quotes differently to most other applications). This makes it tricky to do anything truly universal for all shells. If possible, it's generally better to call executables directly rather than going through a random, unknown, shell.

To be clear, I'm not arguing that nushell shouldn't aim to work here. Just that this is complicated.
